doc/source/image_description/elements.rst

```diff
@@ -572,6 +572,15 @@ overlayroot="true|false"
   will not have any effect because the write partition will be
   resized on first boot to the available disk space.
 
+overlayroot_write_partition="true|false"
+  For the `oem` type only, allows to specify if the extra read-write
+  partition in an `overlayroot` setup should be created or not.
+  By default the partition is created and the kiwi-overlay dracut
+  module also expect it to be present. However, the overlayroot
+  feature can also be used without an initrd and under certain
```

Comment:
I don't understand this explanation 😓 How is an OEM image wihout an initrd? And where is the relation between having or not an initrd and the use of a persistent partitions for overlay deployments?

I am failing to understand potential use cases.

Reply:
ok sorry, let me explain the use case first. See the following type definition

```xml
<type image="oem" filesystem="ext2" firmware="custom" initrd_system="none" overlayroot="true" overlayroot_write_partition="false" bootpartition="true" bootpartsize="20">
    <bootloader name="grub2"/>
    <size unit="M">128</size>
    <partitions>
        <partition name="app" size="10" mountpoint="/app" filesystem="ext2"/>
    </partitions>
</type>
```

This is the type setup of an image with a read-only rootfs based on squashfs. The only persistent write area is a special partition for the application. In kiwi we support squashfs root via the ```overlayroot``` attribute. Along with this overlayroot feature there is always an extra partition that can be used to make the system writable through overlayfs. This extra partition is unwanted on production. The new attribute ```overlayroot_write_partition="true|false"``` allows to control if
this extra write partition for the overlay exists or not.

This is the use case.

If the changelog comment doesn't make this clear it would be great if you can rephrase it such that it makes sense for you. I'm starring at this for tool long and maybe couldn't find the right words to explain it :)

Thanks

Reply:
Also to mention the system boots up with a custom kernel. No initrd is required to bootup as bootup times matter and all drivers are compiled in. Think embedded ;)

Reply:
I see that sounds cool, relatively similar to what I am doing nowadays in terms of setting an overlayfs over an immutable rootfs. 

Out of curiosity how this partition setup works together with  `overlayroot="true"`. I'd say the result is one partition of 10Mb and then a squashfs image dd'ed to a second partition including an fstab line to mount `/app` from the previous partition. I am getting it right? That's fun.

Reply:
yes on  a block layer this looks like this

```
BusyBox v1.30.1 (Ubuntu 1:1.30.1-7ubuntu2) built-in shell (ash)
Enter 'help' for a list of built-in commands.

# lsblk 
NAME   MAJ:MIN RM  SIZE RO TYPE MOUNTPOINTS
vda    254:0    0  128M  0 disk 
|-vda1 254:1    0   20M  0 part
|-vda2 254:2    0   10M  0 part /app
`-vda3 254:3    0   93M  0 part /
```

Only /app is a writable area and the application is prepared for this condition. I'd say it's not possible in a generic way because most linux applications assumes they can write. Fortunately in this world and where it runs this is not an issue. So all on / is read-only and immutable. That also means you can switch off/on the hardware, fast reboot/crash-reboot
cycles were also important. The extra 20MB is the boot partition but it never gets mounted. The system boots up via uboot and that's the only place the kernel gets read from the disk. Unfortunately uboot does fatload so you have to have a fat partition which is that 20M extra space. During development there is an additional partition which allows to boot up the system with overlayroot in RW mode. This is very handy to test/debug and so on. On Production I just add the attribute from this PR and the extra write space is gone